src/main/java/org/spongepowered/common/registry/type/entity/GameModeRegistryModule.java

```diff
@@ -47,7 +47,7 @@ protected void generateInitialMap() {
 
     public static GameType toGameType(GameMode gamemode) {
         for (GameType gameType : GameType.values()) {
-            if (gameType.getName().equalsIgnoreCase(gamemode.getId())) {
+            if (((GameMode) (Object) gameType).getId().equals(gamemode.getId())) {
```

Comment:
Since Sponge implements `GameMode` as a [mixin](https://github.com/SpongePowered/SpongeCommon/blob/9d594880e16905f61784f37d0bc890da05981caf/src/main/java/org/spongepowered/common/mixin/core/data/types/MixinGameType.java#L43) to `GameType`, this method could actually be rewritten to simply `return (GameType) (Object) gamemode;`